spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,210 @@
+ */
+
+package co.cask.hydrator.plugin.spark;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import joptsimple.internal.Strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.Path;
+
+/**
+ * Spark plugin Utility class. Contains common code to be used in trainer and predictors.
+ */
+public class SparkUtils {
+
+  /**
+   * Validate the config parameters for the spark sink and spark compute classes.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param labelField        field containing the prediction values.
+   */
+  public static void validateConfigParameters(Schema inputSchema, String featuresToInclude,
+                                              String featuresToExclude, String labelField) {
+    if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+      throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                           "Please specify fields for one");
+    }
+    List<String> fields = getFeatureList(inputSchema, featuresToInclude, featuresToExclude, labelField);
+    for (String field : fields) {
+      Schema.Field inputField = inputSchema.getField(field);
+      Schema schema = inputField.getSchema();
+      Schema.Type features = schema.isNullableSimple() ? schema.getNonNullable().getType() : schema.getType();
+      if (features.equals(Schema.Type.BYTES) || features.equals(Schema.Type.STRING) ||
+        features.equals(Schema.Type.BOOLEAN)) {
+        throw new IllegalArgumentException(String.format("Field to classify must be continuous : int, double, " +
+                                                           "float, long but was %s.", features));
+      }
+      Preconditions.checkArgument(!features.equals(Schema.Type.NULL), "Field to classify must not be of type null");
+    }
+
+  }
+
+  /**
+   * Get the feature list of the features that have to be used for training/prediction depending on the
+   * featuresToInclude or featuresToInclude list.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param labelField        field containing the prediction values.
+   * @return feature list to be used for training/prediction.
+   */
+  public static List<String> getFeatureList(Schema inputSchema, String featuresToInclude, String featuresToExclude,
```

Comment:
Why you don't need to check the include and exclude features as the one did above?